components/Sections/NewHeader/__tests__/index.test.tsx

```diff
@@ -0,0 +1,128 @@
+import userEvent from '@testing-library/user-event';
+import { IntlProvider } from 'react-intl';
+import { render, fireEvent, screen } from '@testing-library/react';
+import Header from '..';
+import { useMediaQuery } from '../../../../hooks/useMediaQuery';
+
+let mockCurrentTheme = '';
+
+const mockToggleTheme = jest.fn().mockImplementation(() => {
+  mockCurrentTheme = mockCurrentTheme === 'dark' ? 'light' : 'dark';
+});
+
+// Mock dark mode module for controlling dark mode HOC behaviour
+jest.mock('next-themes', () => ({
+  useTheme: () => {
+    return { theme: mockCurrentTheme, setTheme: mockToggleTheme };
+  },
+}));
+
+// mock useRouter
+jest.mock('next/router', () => ({
+  useRouter() {
+    return {
+      locale: 'en',
+    };
+  },
+}));
+
+jest.mock('../../../../hooks/useLocale', () => ({
+  useLocale: () => ({
+    currentLocale: { code: 'en', name: 'English', localName: 'English' },
+    availableLocales: [
+      { code: 'en', name: 'English', localName: 'English' },
+      { code: 'es', name: 'Spanish', localName: 'Español' },
+    ],
+  }),
+}));
+
+// mock useMediaQuery hook for emulating mobile device
+jest.mock('../../../../hooks/useMediaQuery', () => ({
+  useMediaQuery: jest.fn(),
+}));
+
+describe('Tests for Header component', () => {
+  beforeEach(() => {
+    // @ts-ignore
+    useMediaQuery.mockReturnValue(false);
+  });
+
+  afterEach(() => {
+    mockToggleTheme.mockClear();
+    mockCurrentTheme = 'dark';
+  });
+
+  it('renders correctly', () => {
+    const { container } = render(
+      <IntlProvider locale="en" onError={() => {}}>
+        <Header />
+      </IntlProvider>
+    );
+    expect(container).toMatchSnapshot();
```

Comment:
Now we use storybook to create snapshot. you can find information [here](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md#unit-tests-and-storybooks)

Reply:
There is there a storybook snapshot for the Header, so do you mean this test can be removed? Or at least changed to just make sure it renders?

Reply:
yeah we can remove this test. (and use storybook)
But don't miss to unit test the hook

Reply:
Hook tests were added

Reply:
And snapshot was removed.